sourmash/index.py

```diff
@@ -0,0 +1,103 @@
+"An Abstract Base Class for collections of signatures."
+
+from abc import ABCMeta, abstractmethod
+from collections import namedtuple
+
+# @CTB copied out of search.py to deal with import order issues, #willfix
+SearchResult = namedtuple('SearchResult',
+                          'similarity, match_sig, md5, filename, name')
+
+
+# compatible with Python 2 *and* 3:
+ABC = ABCMeta("ABC", (object,), {"__slots__": ()})
+
+
+class Index(ABC):
+    @abstractmethod
+    def find(self, search_fn, *args, **kwargs):
+        """ """
+
+    @abstractmethod
+    def search(self, signature, *args, **kwargs):
```

Comment:
Or even defining `search` like this:
```python
def search(self, signature, *, threshold=None, do_containment=False, ignore_abundance=False, **kwargs):
```
without `*args`, with default keyword arguments.

(the `*` in the middle is a Python 3 thing, but when this get merged I think we will have dropped Python 2 anyway?)